src/nomad_material_processing/general.py

```diff
@@ -0,0 +1,1336 @@
+        a_eln=ELNAnnotation(
+            component='NumberEditQuantity',
+            defaultDisplayUnit='millimeter ** 2',
+        ),
+        unit='meter ** 2',
+    )
+    cap_surface_area = Quantity(
+        type=float,
+        description='Area of the cap.',
+        a_eln=ELNAnnotation(
+            component='NumberEditQuantity',
+            defaultDisplayUnit='millimeter ** 2',
+        ),
+        unit='meter ** 2',
+    )
+    lateral_surface_area = Quantity(
+        type=float,
+        description='Area of the lateral surface.',
+        a_eln=ELNAnnotation(
+            component='NumberEditQuantity',
+            defaultDisplayUnit='millimeter ** 2',
+        ),
+        unit='meter ** 2',
+    )
+
+
+class CylinderSector(Cylinder):
+    central_angle = Quantity(
+        type=float,
+        description="""The angle that defines the portion of the cylinder.
+        This angle is taken at the center of the base circle
+        and extends to the arc that defines the cylindrical sector.""",
+        a_eln=ELNAnnotation(
+            component=ELNComponentEnum.NumberEditQuantity,
+        ),
+        unit='degree',
+    )
+
+
+class IrregularParallelSurfaces(Geometry):
+    """
+    A shape that does not fit into any of the other geometry classes.
+    """
+
+    m_def = Section()
+    height = Quantity(
+        type=float,
+        description='The z dimension of the irregular shape.',
+        a_eln=ELNAnnotation(
+            component='NumberEditQuantity',
+            defaultDisplayUnit='nanometer',
+        ),
+        label='Height (z)',
+        unit='meter',
+    )
+
+
+class Miscut(ArchiveSection):
+    """
+    The miscut in a crystalline substrate refers to
+    the intentional deviation from a specific crystallographic orientation,
+    commonly expressed as the angular displacement of a crystal plane.
+
+    It can be either directed along one of the crystallographic axes or in a specific
+    direction in the crystal defined by a main direction and the perpendicular one.
+    """
+
+    angle = Quantity(
+        type=float,
+        description="""
+        The angular displacement offset toward the crystallographic
+        orientation of the substrate.
+        """,
+        a_eln=ELNAnnotation(
+            component=ELNComponentEnum.NumberEditQuantity,
+            defaultDisplayUnit='deg',
+            label='Miscut Angle',
+        ),
+        unit='deg',
+    )
+    angle_deviation = Quantity(
+        type=float,
+        description='The ± deviation of the angular displacement offset.',
+        a_eln=ELNAnnotation(
+            component=ELNComponentEnum.NumberEditQuantity,
+            defaultDisplayUnit='deg',
+            label='± Miscut Angle Deviation',
+        ),
+        unit='deg',
+    )
+    hkl_absolut_orientation = Quantity(
+        type=str,
+        description="""Direction where the miscut goes toward.
+        Here the absolut Miller indices, (hkl), denote a normal to the planes
```

Comment:
> Here the absolut Miller indices, (hkl)

1. Typo in _absolute_.
2. hkl cover all crystal lattices, but for some (hexagonal and rhombohedral) it's a popular choice to add a fourth index. I haven't worked much with those myself, though I can take a look there too. I think @JosePizarro3 has more experience with hexagonal lattices.

Reply:
I just had a confirmation from some other guys. They say that when you have 4 indices, the third is pleonastic and can be derived by the 1st, 2nd, and 4th.

For now, we do not insert it, we might think to a subclass for those cells